src/TLabel.cpp

```diff
@@ -114,3 +146,60 @@ void TLabel::enterEvent(QEvent* event)
     }
     QWidget::enterEvent(event);
 }
+
+bool TLabel::forwardEventToMapper(QEvent* event)
+{
+    if (event->type() == QEvent::MouseButtonPress || event->type() == QEvent::MouseButtonRelease || event->type() == QEvent::MouseMove)
+    {
+        QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
```

Comment:
Given that we are dealing with `QObject` derived types here (we are, aren't we?) a `qobject_cast` may be safer here than a `static_cast` - together with the mandatory test for `mouseEvent` not being a `nullptr`...!

Reply:
It doesn't appear that the QEvent class is derived from QObject. I'll definitely add the nullptr check though!